src/Compilers/CSharp/Test/Semantic/Semantics/PatternMatchingTests_ListPatterns.cs

```diff
@@ -0,0 +1,1325 @@
+{
+    public static void Main()
+    {
+        _ = new Test1() is {0};
+    } 
+}
+";
+            var compilation = CreateCompilationWithIndexAndRange(source, parseOptions: TestOptions.RegularWithListPatterns, options: TestOptions.ReleaseExe);
+            compilation.VerifyEmitDiagnostics(
+                // (12,28): error CS9200: List patterns may not be used for a value of type 'Test1'.
+                //         _ = new Test1() is {0};
+                Diagnostic(ErrorCode.ERR_UnsupportedTypeForListPattern, "{0}").WithArguments("Test1").WithLocation(12, 28));
+        }
+
+        [Fact]
+        public void ListPattern_MemberLookup_OverridenIndexer()
+        {
+            var source = @"
+using System;
+class Test1
+{
+    public virtual int this[Index i] => 1;
+    public int Count => 0;
+}
+class Test2 : Test1
+{
+}
+class Test3 : Test2
+{
+    public override int this[Index i] => 1;
+}
+class X
+{
+    public static void Main()
+    {
+        _ = new Test1() is {0};
+        _ = new Test2() is {0};
+        _ = new Test3() is {0};
+    } 
+}
+";
+            var compilation = CreateCompilationWithIndexAndRange(source, parseOptions: TestOptions.RegularWithListPatterns, options: TestOptions.ReleaseExe);
+            compilation.VerifyEmitDiagnostics();
+        }
+
+
+        [Fact]
+        public void ListPattern_RefReturns()
+        {
+            var source = @"
+using System;
+class Test1
+{
+    int value = 1;
+    public ref int this[Index i] => ref value;
+    public int Count => 1;
+}
+class X
+{
+    public static void Main()
+    {
+        Console.Write(new Test1() is {1});
+    } 
+}
+";
+            var compilation = CreateCompilationWithIndexAndRange(source, parseOptions: TestOptions.RegularWithListPatterns, options: TestOptions.ReleaseExe);
+            compilation.VerifyEmitDiagnostics();
+            var verifier = CompileAndVerify(compilation, expectedOutput: "True");
+            verifier.VerifyIL("X.Main", @"
+{
+  // Code size       44 (0x2c)
+  .maxstack  3
+  .locals init (Test1 V_0)
+  IL_0000:  newobj     ""Test1..ctor()""
+  IL_0005:  stloc.0
+  IL_0006:  ldloc.0
+  IL_0007:  brfalse.s  IL_0025
+  IL_0009:  ldloc.0
+  IL_000a:  callvirt   ""int Test1.Count.get""
+  IL_000f:  ldc.i4.1
+  IL_0010:  bne.un.s   IL_0025
+  IL_0012:  ldloc.0
+  IL_0013:  ldc.i4.0
+  IL_0014:  ldc.i4.0
+  IL_0015:  newobj     ""System.Index..ctor(int, bool)""
+  IL_001a:  callvirt   ""ref int Test1.this[System.Index].get""
+  IL_001f:  ldind.i4
+  IL_0020:  ldc.i4.1
+  IL_0021:  ceq
+  IL_0023:  br.s       IL_0026
+  IL_0025:  ldc.i4.0
+  IL_0026:  call       ""void System.Console.Write(bool)""
+  IL_002b:  ret
+}");
+        }
+
+        [Fact]
+        public void SlicePattern_SliceValue()
+        {
+            var source = @"
+using System;
+class X
+{
+    public static void Main()
+    {
+        var arr = new[] { 1, 2, 3, 4 };
+        if (arr is {.. var start, _})
+            Console.Write(string.Join("""", start));
```

Comment:
nit: Did you mean to have a space in the `Join`? I think that would help  with readability of the test